cart-service/src/cart/services/cart.service.ts

```diff
@@ -34,22 +46,58 @@ export class CartService {
     return this.createByUserId(userId);
   }
 
-  updateByUserId(userId: string, { items }: Cart): Cart {
-    const { id, ...rest } = this.findOrCreateByUserId(userId);
-
-    const updatedCart = {
-      id,
-      ...rest,
-      items: [ ...items ],
+  async updateByUserId(userId: string, item: LegacyCartItem): Promise<any> {
+    const { id, ...rest } = await this.findOrCreateByUserId(userId);
+    const cart = await this.findOrCreateByUserId(userId);
+    console.log('cart', cart);
+
+    let cartItem = cart.items.find(
+      (record) => record.productId === item.product.id,
+    );
+    if (cartItem) {
+      if (item.count === 0) {
+        await this.cartItemRepository.remove(cartItem);
```

Comment:
Probably, a 'return' statement is missing here

Reply:
Correct. Thanks